src/event-store.ts

```diff
@@ -379,7 +379,7 @@ export class EventStore<Q> {
   public async listen<T extends EventData>(
     event_namespace: T["event_namespace"],
     event_type: T["event_type"],
-    handler: EventHandler<Q, any>,
+    handler: EventHandler<Q, Event<T, any>>,
```

Comment:
Could this be made even better with `EventContext`?

```suggestion
    handler: EventHandler<Q, Event<T, EventContext<any>>>,
```